run_selftest.py

```diff
@@ -0,0 +1,123 @@
+                    '--log', f'sumary_log.html']
+    for component in COMPONENTS:
+        outfile = run_cmpt_selftest(component)
+        rebot_args.append(outfile)
+    # merge Component results
+
+    res = subprocess.run([*REBOT_CMD, *rebot_args], capture_output=True)
+    print(f"Sumary report: {output_dir}")
+        
+def run_cmpt_selftest(cmpt):
+    
+    if cmpt == "variants":
+        # output_dir = os.path.join(BASE_DIR, 'testcases', 'output')
+        output_dir = os.path.join(BASE_DIR, 'output')
+        rebot_args = ['--name', cmpt,
+                      '--outputdir', output_dir, 
+                      '--output', f'{cmpt}.xml',
+                      '--report', f'{cmpt}_report.html',
+                      '--log', f'{cmpt}_log.html']
+        for variant in VARIANTS:
+            
+            outfile = run_robot(cmpt, variant)
+            
+            rebot_args.append(outfile)
+
+        # merge Variant results
+        res = subprocess.run([*REBOT_CMD, *rebot_args], capture_output=True)
+        # outfile = os.path.join(BASE_DIR, 'testcases', 'out', f'{cmpt}_output.xml')
+        outfile = os.path.join(output_dir, f'{cmpt}.xml')
+    else:
+        outfile = run_robot(cmpt)
+
+    return outfile
+
+def run_robot(cmpt, variant=None):
+    variable = ''
+    suitename = cmpt
+    # output = os.path.join(BASE_DIR, 'testcases', 'out', f'{suitename}.xml') 
+    output_dir = os.path.join(BASE_DIR, 'output') 
+    if variant:
+        variable = f'variant:{variant}'
+        suitename  = f'{cmpt}.{variant}'
+        # output   = os.path.join(BASE_DIR, 'testcases', cmpt, 'out', f'{suitename}.xml') 
+        output_dir = os.path.join(BASE_DIR, 'output', cmpt)
+        if variant == 'default':
+            variable = ''
+    else:
+        variable = f'variant:{OS_VARIANT}'
+
+    robot_target = os.path.join(BASE_DIR, 'testcases', cmpt)
+    print(f'>>> Execute {suitename} ...')
+
+    vars = []
+    if variable:
+        vars.extend(['--variable', variable])
+
+    res = subprocess.run([*ROBOT_CMD, '--name', suitename, *vars,
+    # res = subprocess.Popen([*ROBOT_CMD, '--name', suitename, *vars,
+                        '--outputdir', output_dir, 
+                        '--output', f'{suitename}.xml', 
+                        '--report', f'{cmpt}_report.html',
+                        '--log', f'{cmpt}_log.html', 
+                        # robot_target], capture_output=True)
+                        robot_target])
+
+    outfile = os.path.join(output_dir, f'{suitename}.xml')
+    print(outfile)
+    return outfile
+
+def add_selftest_ext():
+    print("Adding selftest-ext testcases ...")
+    shutil.copytree(os.path.join(SELFTEST_EXT_DIR, 'testcases'), TESTCASE_DIR, dirs_exist_ok=True)
+
+    print("Adding selftest-ext configurations ...")
+    shutil.copytree(os.path.join(SELFTEST_EXT_DIR, 'config'), os.path.join(BASE_DIR, 'config'), dirs_exist_ok=True)
+
+
+if __name__ == "__main__":
+
+    SELFTEST_EXT_DIR = ""
```

Comment:
Hi Ngoan,
I think this problem is solved by means of our new test-trigger and append-mode of robotlog2db.
Important is that selftest-extension can be executed independently of selftest.
What is your opinion?
Thank you,
 Thomas

Reply:
Hi Thomas,

My initial idea is the extension only contains the additional testcases and extended configurations (internal Bosch), the test executer (**run_selftest.py**) and main configuration is hosted in this repo.
So that I implemented to copy all additional testcase and configuration into this selftest folder before triggering Robot testcases execution to get the single result (log, report) file for all and avoid duplication code.

But I got the idea from you and Holger with test trigger and append mode in our meeting then I have removed this implementation (commented `add_selftest_ext()` at line 115) and reworked in the extension repo accordingly.

Thank you,
Ngoan